src/components/Charts/LineChart.tsx

```diff
@@ -0,0 +1,120 @@
+ * However, the preferred type is `LineChartData`.
+ *
+ * To enable the LineChart component to support an additional `data` type, the following steps are required:
+ * 1. Define an interface and add it to `LineChartProps.data`.
+ * 2. Add another switch case in `LineChartOperations.convertToLineChartData` to convert the new interface to `LineChartData`.
+ *
+ * @param title chart title (optional)
+ * @param description chart description text (optional)
+ * @param expandable when selected, the user is given the option to open the chart in a larger modal (optional)
+ * @param small when selected, all components in the line chart box become slightly smaller (optional)
+ * @param data the actual data to be used in the chart
+ */
+export function LineChart({ title, description, expandable, small, data }: LineChartProps) {
+  const TITLE_TEXT_SIZE = small ? 'text-sm' : 'text-md';
+  const DESCRIPTION_TEXT_SIZE = small ? 'text-xs' : 'text-sm';
+  const CHART_HEIGHT = small ? 12 : 16;
+  const ICON_BUTTON_SIZE = small ? 3 : 4;
+  const HEADER_BOTTOM_PADDING = title ? 3 : 0;
+  const JSON_DOWNLOAD_FILE_NAME = `hunger_map_line_chart_json-${title}.json`;
+
+  // full screen modal state handling
+  const { isOpen, onOpen, onOpenChange } = useDisclosure();
+
+  // convert data to `LineChartData` and build chart options for 'Highcharts'
+  const lineChartData: LineChartData = LineChartOperations.convertToLineChartData(data);
+  const chartOptions: Highcharts.Options = LineChartOperations.getHighChartData(lineChartData);
+
+  // trigger download of the given line chart `data` as a json file
+  const downloadDataJson = () => {
+    // convert data json object to string and encode as URI
+    const jsonString = `data:text/json;charset=utf-8,${encodeURIComponent(JSON.stringify(data, null, 2))}`;
+    // create a temporary link element and trigger the download
+    const link = document.createElement('a');
+    link.href = jsonString;
+    link.download = JSON_DOWNLOAD_FILE_NAME;
+    link.click();
+  };
+
+  // Full screen modal that can be opened if `expandable==true`. Offers a larger chart and a download button.
+  const fullScreenModal = (
+    <Modal size="5xl" isOpen={isOpen} backdrop="blur" scrollBehavior="inside" onOpenChange={onOpenChange}>
+      <ModalContent>
+        <ModalHeader className="flex flex-col gap-1">{title}</ModalHeader>
+        <ModalBody>
+          <p className="w-full h-fit text-md font-normal">{description}</p>
+          <div className="py-6">
+            <HighchartsReact
+              highcharts={Highcharts}
+              options={chartOptions}
+              containerProps={{ style: { width: '100%', height: '40vh', borderRadius: '0 0 0.5rem 0.5rem' } }}
+            />
+          </div>
+        </ModalBody>
+        <ModalFooter>
+          <Button color="primary" onPress={downloadDataJson}>
+            Download data as JSON
+          </Button>
+        </ModalFooter>
+      </ModalContent>
+    </Modal>
+  );
```

Comment:
Can we not use your popup component? Anyways this component should go into separate file called ExpandedChart or something like this